### ISSUE TITLE
Add `required` field for empty parameters

### ISSUE THREAD

**Issue by domire8** [aica-technology/api#99]
https://github.com/aica-technology/api/blob/c94ebae55013754c2533d551633abe1588c719b8/schemas/parameters/schema/parameter.schema.json#L1

We have components and controllers that have parameters which don't have a default value but need to be set such that the controller/component can run correctly. Is that something we should add to the schema such that we can visualize it in the frontend? 

What happened to me is that I added a timer and tried to run it without setting the `timeout` parameter (which by the way obivously shows `default` next to its empty value - which is an additional point of confusion maybe) and then I wondered why its not running.

I know it's a detail and quite hard to be strict about it because parameters can have no default value and not be required because the component handles that internally but they can also have empty parameters where we don't know what a sensible value is and then the user needs to set it.

Long story short, I think we can improve the experience here, either by tackling it
- through a more detailed schema
- or by having stricter parameter guidelines on the controller/component side where we say parameters cannot be empty if they are required to run the controller/component.

What are your thoughts @eeberhard ?

**Comment by eeberhard** [comment]
The design intent was for `default: null` to indicate that a value is required. From [the docs](https://docs.aica.tech/docs/reference/custom-components/component-descriptions#parameters):

> When a parameter has no valid default state and must be set by the user for the component to function, the default_value property in the component description should be set to null. This is distinct from a valid default empty parameter state, which can be expressed with the empty string "".

But I agree it's not immediately clear, and certainly the UI doesn't distinguish this.

The reasoning is that a parameter that doesn't have a default value _must_ be set, therefore is required. Do you think there are any reasonable cases where you would say a parameter is _required_ even though it has default value? What about a parameter that doesn't have a default value but is also optional?

The problem with two fields is that we have four possible combinations; required with default value, required with no default, not required with default value, not required with no default value. Of those I think only "required with no default, not required with default value" make sense, in which case there is no point to have a `required` field.

For me it seems more like a documentation and UI issue to make required parameters more obvious. If there is no default value, we can highlight that parameter in the UI (instead of just showing it as the current "default")

**Comment by domire8** [comment]
> Do you think there are any reasonable cases where you would say a parameter is required even though it has default value? 

I think no, that would be bad design

> What about a parameter that doesn't have a default value but is also optional?

That can happen, for example the reference pose and angular limit in the IK controller. If not set, we don't do anything with it.

I think the 'not required with no default value' is therefore also a valid case but no strong feelings..

**Comment by eeberhard** [comment]
Ok, as long as there is one reasonable use case then it's worth adding. But considering the case again, the property should be called `optional` and not `required`, since only the "optional, null default" makes sense but "required, null default" does not.

The property `optional` can be added to the parameter schema with the following rules:
- If default value is null, the schema _may_ include an `optional` boolean property (but this property is not required)
- If default value is null and `optional` is not specified, optional is assumed to be false (user parameter value required)
- If default value is null and `optional` _is_ specified, user parameter value is or is not required depending on the state of `optional`
- If default value is not null and `optional` _is_ specified, it should be a schema error.
